gips/data/landsat/landsat.py

```diff
@@ -329,17 +329,32 @@ def fetch(cls, asset, tile, date):
         if len(response) > 0:
             verbose_out('Fetching %s %s %s' % (asset, tile, fdate), 1)
             if len(response) != 1:
-                raise Exception('Single date, single location, returned more than one result')
+                raise Exception('Single date, single location, '
+                                'returned more than one result')
             result = response[0]
+            utils.verbose_out(str(response), 4)
             sceneID = result['sceneID']
-            stage_dir = os.path.join(cls.Repository.path(), 'stage')
+            stage_dir = cls.Repository.path('stage')
             sceneIDs = [str(sceneID)]
 
             username = settings().REPOS['landsat']['username']
             password = settings().REPOS['landsat']['password']
             api_key = api.login(username, password)['data']
-            url = api.download(result['dataset'], 'EE', sceneIDs, 'STANDARD', api_key)['data'][0]
-            download(url, stage_dir)
+            url = api.download(
+                result['dataset'], 'EE', sceneIDs, 'STANDARD', api_key
+            )['data'][0]
+            with utils.make_temp_dir(prefix='dwnld', dir=stage_dir) as dldir:
+                download(url, dldir)
+                granules = os.listdir(dldir)
+                if len(granules) != 1:
+                    raise Exception(
+                        'Download appears to have not produced a file: {}'
+                        .format(str(granules))
```

Comment:
It looks like when `len(granules) > 1`, the exception may have a confusing message.  Is it worth it to reword it?  Maybe:  `"Expected 1 thing, but got {} things: {things}".format(len(things), things)`

Also I believe `"{}".format(some_list)` does the `str()` part for you if you do decide to change this bit.

Reply:
A few lines back the code ensures that we only got one URL from the query_service, so provided `download(one_url)` implies we get  1 file, then the only case remaining is the no-file case.  Does that make sense.  I guess I will change it to `if not granules:`.  Seem safer?

Reply:
err...I'll go `len(granules) == 0` for clarity.